src/AspNet.Security.OAuth.DingTalk/DingTalkAuthenticationHandler.cs

```diff
@@ -104,50 +108,44 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OA
 
         protected override string FormatScope() => string.Join(",", Options.Scope);
 
+        private async Task<JsonDocument> CopyPayloadAsync(Dictionary<string, StringValues> content)
+        {
+            var bufferWriter = new ArrayBufferWriter<byte>();
+
+            await using (var writer = new Utf8JsonWriter(bufferWriter))
+            {
+                writer.WriteStartObject();
+
+                foreach (var item in content)
+                {
+                    writer.WriteString(item.Key, item.Value);
+                }
+
+                writer.WriteEndObject();
+                await writer.FlushAsync();
+            }
+
+            return JsonDocument.Parse(bufferWriter.WrittenMemory);
```

Comment:
This JSON document is just being generated to re-parse it and then turn it into a string to then pass to a `StringContent` instance.

If you returned the `ReadOnlyMemory<T>` from the `WrittenMemory` property instead you could then just write it into a [`ReadOnlyMemoryContent`](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.readonlymemorycontent?view=netcore-3.1) instead and save the additional copying and memory allocations.

Reply:
Has been adjusted